+sw_tests/+unit_tests/unittest_ndbase_optimisers.m

```diff
@@ -48,28 +57,27 @@ function test_optimise_rosen_upper_bound_minimum_accessible(testCase, optimiser)
         end
 
         function test_optimise_rosen_upper_bound_minimum_not_accessible(testCase, optimiser)
-            % note intital guess is outside bounds
             [pars_fit, cost_val, ~] = optimiser([], testCase.rosenbrock, [-1,-1], 'ub', [0, inf]);
             testCase.verify_val(pars_fit, [0, 0], 'abs_tol', 1e-3);
-            testCase.verify_val(cost_val, 1, 'abs_tol', 1e-6);
+            testCase.verify_val(cost_val, 1, 'abs_tol', 2e-3);
```

Comment:
It is worse than the `simplex` method - I think because of the effect of bounds on the finite-difference and associated convergence criteria as discussed in another comment. But I think this is a problem for all LM algorithms: the performance is comparable to MATLAB's  `lsqnonlin`
```
rosenbrock = @(x) (1-x(1)).^2 + 100*(x(2) - x(1).^2).^2; 

[pars_lsq, cost_lsq] = lsqnonlin(rosenbrock,[-1, -1], [], [0 inf]);
pars_lsq  %  [ 0    0.0026]
cost_lsq % 1.0014

[pars_lm4, cost_lm4, ~] = ndbase.lm4([],rosenbrock,[-1,-1], 'ub', [0, inf]);
pars_lm4 % [-5.6171e-04 -3.3115e-04]
cost_lm4 % 1.0011
```

Though it should be noted the minimum is on one of the upper parameter bounds, which is perhaps a bit of a challenge!

Reply:
To some extent this can be improved (for both algorithms) by changing the default parameters - but I did play around a lot with the `lm4` defaults on NIST data sets and I think they strike a good balance between speed and accuracy - the user is free to disagree and change them!